### PR TITLE
Updates to support Verizon Ads SDK v1.1.1

### PR DESCRIPTION
**Pull request by nimrich**

### CODE REVIEW

File: Verizon/src/main/java/com/mopub/mobileads/VerizonInterstitial.java [hunk 1]
```diff
@@ -69,39 +71,37 @@ protected void loadInterstitial(final Context context,
...
                 logAndNotifyInterstitialFailed(LOAD_FAILED, ADAPTER_CONFIGURATION_ERROR);
-
-                return;
             }
         }
 
-        final String placementId = serverExtras.get(getPlacementIdKey());
+        // Ensure that siteId is the key and cache serverExtras so siteId can be used to initialize VAS early at next launch
+        if (!TextUtils.isEmpty(siteId)) {
+            serverExtras.put(VerizonAdapterConfiguration.VAS_SITE_ID_KEY, siteId);
```
Comment (chauduyphanvu1) [inline Verizon/src/main/java/com/mopub/mobileads/VerizonInterstitial.java]: Isn't the site ID String returned from the MoPub ad server? If so, it's always going to be in `serverExtras` that you can cache (by calling `verizonAdapterConfiguration.setCachedInitializationParameters(context, serverExtras)`) right away. There's no need to put another entry for site ID in the Map.

File: Verizon/src/main/java/com/mopub/mobileads/VerizonInterstitial.java [hunk 2]
```diff
@@ -69,39 +71,37 @@ protected void loadInterstitial(final Context context,
...
+                application = (Application) context;
+            } else if (context instanceof Activity) {
+                application = ((Activity) context).getApplication();
             }
 
-            final boolean success = StandardEdition.initializeWithActivity((Activity) context, siteId);
 
-            if (!success) {
-                MoPubLog.log(CUSTOM, ADAPTER_NAME, "Failed to initialize the Verizon SDK");
+            if ((application != null) && (!StandardEdition.initialize(application, siteId))) {
```
Comment (chauduyphanvu1) [inline Verizon/src/main/java/com/mopub/mobileads/VerizonInterstitial.java]: Do you mean to check if `application == null` here? Also, remove extra parentheses.

File: Verizon/src/main/java/com/mopub/mobileads/VerizonInterstitial.java [hunk 3]
```diff
@@ -69,39 +71,37 @@ protected void loadInterstitial(final Context context,
...
-
-            if (!(context instanceof Activity)) {
-                MoPubLog.log(CUSTOM, ADAPTER_NAME, "Ad request to Verizon failed because " +
-                        "context is not an Activity");
-
-                logAndNotifyInterstitialFailed(LOAD_FAILED, ADAPTER_CONFIGURATION_ERROR);
-
-                return;
+        if (!VASAds.isInitialized()) {
+            Application application = null;
```
Comment (chauduyphanvu1) [inline Verizon/src/main/java/com/mopub/mobileads/VerizonInterstitial.java]: Add new line below.

File: Verizon/src/main/java/com/mopub/mobileads/VerizonBanner.java [hunk 4]
```diff
@@ -75,39 +77,34 @@ protected void loadBanner(final Context context,
...
-            if (!success) {
-                MoPubLog.log(CUSTOM, ADAPTER_NAME, "Failed to initialize the Verizon SDK");
-
-                logAndNotifyBannerFailed(LOAD_FAILED, ADAPTER_CONFIGURATION_ERROR);
-
-                return;
-            }
+        // Ensure that siteId is the key and cache serverExtras so siteId can be used to initalizate VAS early at next launch
+        if (!TextUtils.isEmpty(siteId)) {
+                serverExtras.put(VerizonAdapterConfiguration.VAS_SITE_ID_KEY, siteId);
```
Comment (chauduyphanvu1) [inline Verizon/src/main/java/com/mopub/mobileads/VerizonBanner.java]: Same comment regarding putting another site ID entry to this Map.

File: Verizon/src/main/java/com/mopub/mobileads/VerizonBanner.java [hunk 5]
```diff
@@ -75,39 +77,34 @@ protected void loadBanner(final Context context,
...
+            if (context instanceof Application) {
+                application = (Application) context;
+            } else if (context instanceof Activity) {
+                application = ((Activity) context).getApplication();
             }
 
-            if (!(context instanceof Activity)) {
-                MoPubLog.log(CUSTOM, ADAPTER_NAME, "Ad request to Verizon failed because " +
-                        "context is not an Activity");
+            if ((application != null) && (!StandardEdition.initialize(application, siteId))) {
```
Comment (chauduyphanvu1) [inline Verizon/src/main/java/com/mopub/mobileads/VerizonBanner.java]: if `application == null`?

File: Verizon/src/main/java/com/mopub/mobileads/VerizonAdapterConfiguration.java [hunk 6]
```diff
@@ -60,17 +65,30 @@ public void initializeNetwork(@NonNull final Context context, @Nullable final Ma
...
         }
+
+        final String finalSiteId = siteId;
+        ThreadUtils.postOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                if ((!TextUtils.isEmpty(finalSiteId)) && (context instanceof Application) &&
+                    (StandardEdition.initialize((Application) context, finalSiteId))) {
+
+                    listener.onNetworkInitializationFinished(VerizonAdapterConfiguration.class,
```
Comment (chauduyphanvu1) [inline Verizon/src/main/java/com/mopub/mobileads/VerizonAdapterConfiguration.java]: Please also notify `listener.onNetworkInitializationFinished()` with the `MoPubErrorCode.ADAPTER_CONFIGURATION_ERROR` error code when the if conditional on line 77 isn't `true`.

File: Verizon/src/main/java/com/mopub/mobileads/VerizonAdapterConfiguration.java [hunk 7]
```diff
@@ -60,17 +65,30 @@ public void initializeNetwork(@NonNull final Context context, @Nullable final Ma
...
+        if (configuration != null) {
+            siteId = configuration.get(VAS_SITE_ID_KEY);
         }
+
+        final String finalSiteId = siteId;
+        ThreadUtils.postOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                if ((!TextUtils.isEmpty(finalSiteId)) && (context instanceof Application) &&
+                    (StandardEdition.initialize((Application) context, finalSiteId))) {
```
[the remark this answers is not in the document]
Comment (nimrich) [reply Verizon/src/main/java/com/mopub/mobileads/VerizonAdapterConfiguration.java]: It does initialize the SDK.

File: Verizon/src/main/java/com/mopub/mobileads/VerizonBanner.java [hunk 8]
```diff
@@ -75,39 +77,34 @@ protected void loadBanner(final Context context,
...
+        if (!TextUtils.isEmpty(siteId)) {
+                serverExtras.put(VerizonAdapterConfiguration.VAS_SITE_ID_KEY, siteId);
         }
+        verizonAdapterConfiguration.setCachedInitializationParameters(context, serverExtras);
 
-        final String placementId = serverExtras.get(getPlacementIdKey());
+        // The current activity must be set as resumed so VAS can track ad visibility
+        ActivityStateManager activityStateManager = VASAds.getActivityStateManager();
+        if ((activityStateManager != null) && (context instanceof Activity)) {
+                activityStateManager.setState((Activity) context, ActivityStateManager.ActivityState.RESUMED);
```
[the remark this answers is not in the document]
Comment (nimrich) [reply Verizon/src/main/java/com/mopub/mobileads/VerizonBanner.java]: The access level was modified in v1.1.1 of the Verizon SDK.

File: Verizon/src/main/java/com/mopub/mobileads/VerizonInterstitial.java [hunk 9]
```diff
@@ -69,39 +71,37 @@ protected void loadInterstitial(final Context context,
...
+            serverExtras.put(VerizonAdapterConfiguration.VAS_SITE_ID_KEY, siteId);
+        }
+        if (verizonAdapterConfiguration != null) {
+            verizonAdapterConfiguration.setCachedInitializationParameters(context, serverExtras);
+        }
+
+        // The current activity must be set as resumed so VAS can track ad visibility
+        ActivityStateManager activityStateManager = VASAds.getActivityStateManager();
+        if ((activityStateManager != null) && (context instanceof Activity)) {
+            activityStateManager.setState((Activity) context, ActivityStateManager.ActivityState.RESUMED);
```
[the remark this answers is not in the document]
Comment (nimrich) [reply Verizon/src/main/java/com/mopub/mobileads/VerizonInterstitial.java]: The access level was modified in v1.1.1 of the Verizon SDK.

**Comment by nimrich** [comment]
@ chauduyphanvu1 I believe I have addressed all of your comments.  Thanks so much for the quick turnaround on the review!

Just let me know if there's anything else I missed!

File: Verizon/src/main/java/com/mopub/mobileads/VerizonAdapterConfiguration.java [hunk 10]
```diff
@@ -60,17 +65,40 @@ public void initializeNetwork(@NonNull final Context context, @Nullable final Ma
...
+                    listener.onNetworkInitializationFinished(VerizonAdapterConfiguration.class,
+                        MoPubErrorCode.ADAPTER_CONFIGURATION_ERROR);
+                }
+
+                final MoPubLog.LogLevel mopubLogLevel = MoPubLog.getLogLevel();
+
+                if (mopubLogLevel == MoPubLog.LogLevel.DEBUG) {
+                    VASAds.setLogLevel(Logger.DEBUG);
+                } else if (mopubLogLevel == MoPubLog.LogLevel.INFO) {
+                    VASAds.setLogLevel(Logger.INFO);
```
[the remark this answers is not in the document]
Comment (nimrich) [reply Verizon/src/main/java/com/mopub/mobileads/VerizonAdapterConfiguration.java]: It does not need to be initialized first so I'll go ahead and move it.

**Comment by nimrich** [comment]
@chauduyphanvu1 went ahead and made that change.

**Pull request merged**